### PR TITLE
gha: add gs creds

### PR DESCRIPTION
**Pull request by efiop**
Related to #4418

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -52,6 +53,11 @@ jobs:
       run: |
         git config --global user.email "dvctester@example.com"
         git config --global user.name "DVC Tester"
+    - name: setup gs creds
+      shell: bash
+      run: |
+        mkdir -p scripts/ci
```
Comment (efiop) [inline .github/workflows/tests.yaml:59]: Using what is already used in the code for now. Don't want to mess with the tests themselves in this PR, just making this run the same way it did on travis.

**Review by pared — APPROVED** [review]

**Comment by pared** [comment]
Failing test seems unrelated: it is `test_open_granular[azure]`. It seems like it started to fail recently. Will create an issue for that.

File: tests/remotes/gs.py [hunk 2]
```diff
@@ -25,27 +25,13 @@
...
-                [
-                    "gcloud",
-                    "auth",
-                    "activate-service-account",
-                    "--key-file",
-                    TEST_GCP_CREDS_FILE,
-                ]
-            )
-        except (CalledProcessError, OSError):
-            return False
```
Comment (efiop) [inline tests/remotes/gs.py:48]: Had to get rid of this to make gs tests less flaky. This is an old leftover that is no longer needed after test_data_cloud CLI refactoring done a few months back, where we started properly using config.

**Comment by efiop** [comment]
gs mac/linux tests are operational, waiting for windows tests to complete...

**Pull request merged**